src/client.rs

```diff
@@ -0,0 +1,45 @@
+use std::net::UdpSocket;
+use std::sync::Arc;
+
+pub trait ClientTrait {
+    fn get_id(&self) -> u16;
+    fn get_id_byte(&self) -> &[u8];
+    fn read_loop(&mut self);
+}
+
+struct Client {
+    id: u16,
+    id_byte: [u8; 2],
+    sock: Arc<UdpSocket>,
+}
+
+impl Client {
+    pub fn new(id: u16, addr: String) -> Result<Self, anyhow::Error> {
```

Comment:
ここの `addr` の型 `String` じゃなくて [UdpSocket::bind()](https://doc.rust-lang.org/std/net/struct.UdpSocket.html#method.bind) の型に合わせたほうが良いかも

```suggestion
    pub fn new(id: u16, addr: impl ToSocketAddrs) -> Result<Self, anyhow::Error> {
```